themes/default/content/blog/deploy-ai-ml-vercel-app/index.md

```diff
@@ -0,0 +1,190 @@
+If you are ready to integrate your chatbot and deploy it to Vercel along with us, then let's double check a few requirements before we continue. While Pulumi supports many languages, we chose Python for this project, as Python is a familiar language in the AI industry and this is part of the *#MLOpsChallenge*.
+
+### Requirements
+
+* [Pulumi CLI](https://www.pulumi.com/docs/install/)
+* [Pulumi account and access token](https://app.pulumi.com/signup)
+* [Python3](https://www.python.org/downloads/)
+* [Git CLI](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* Endpoint of Katwalk Server API. Follow the steps in [this article](https://www.pulumi.com/blog/mlops-the-ai-challenge-is-cloud-not-code/) to deploy the API and get your endpoint.
+* [Vercel account and access token](https://vercel.com/guides/how-do-i-use-a-vercel-api-access-token#creating-an-access-token)
+
+
+Now that you have everything needed to get started, let's hook up our frontend and deploy it to vercel using Pulumi.
+
+### 1. Clone the repo and setup your project
+
+The first we need to do is clone the repository from GitHub and setup the Pulumi Python environment.
+
+```bash
+# Clone the repository and cd to the infra directory
+git clone https://github.com/pulumiverse/katwalk-frontend && cd katwalk-frontend/infra
+
+# Create and initialize the python virtual env
+python3 -m venv venv && source venv/bin/activate
+
+# Install python dependencies
+python -m pip install -r requirements.txt
+```
+
+### 2. Login to Pulumi Cloud and initialize stack
+
+Next, let's setup our state and secrets store in Pulumi Cloud. 
+
+```bash
+# You can store Pulumi state in multiple ways, here we use Pulumi Cloud
+# Other state backends include Azure storage, Amazon S3, local file, and more
+pulumi login 
+
+# Initialize your stack
+# Here we name the stack "dev"
+pulumi stack init --stack dev
+```
+
+### 3. Configure deployment credentials
+
+Before the deployment, you will need to set the config values for your Vercel token, katwalk-api dns name, repsitory name, and repository type. The Vercel token is supposed to be set as a secret. If you open the `Pulumi.${stackName}.yaml` file, you can see the encrypted secret in this plain text YAML configuration.
+
+```bash
+# Set your katwalk-api path here - It should have this format https://{domain}/v1/chat
+pulumi config set backendDns <katwalk_api>
+
+# The name of the git repository. It should have this format {account-name}/{repository-name}
+pulumi config set repoName <repo_name>
+
+# The git provider of the repository. Must be either `github`, `gitlab`, or `bitbucket`.
+pulumi config set repoType <repo_name>
+
+# Configure vercel token as secret
+pulumi config set --secret token <vercel_token>
+
+```
+
+### 4. Deploy to Vercel
+
+Now, you're set up to deploy your Vue app to Vercel. 
+
+```bash
+# Finally, run `pulumi up` to deploy!
+pulumi up
+
+# When done, you can 'destroy' the stack to deprovision your deployment
```

Comment:
I'd recommend moving this command (destroying the infrastructure) until after the section on interacting with the chatbot using the frontend.